packages/medusa/src/loaders/helpers/routing/index.ts

```diff
@@ -0,0 +1,566 @@
+
+    try {
+      await import(absolutePath).then((imp) => {
+        const middlewaresConfig = imp.config as MiddlewaresConfig | undefined
+
+        if (!middlewaresConfig) {
+          log({
+            activityId: this.activityId,
+            message: `No middleware configuration found in ${absolutePath}. Skipping middleware configuration.`,
+          })
+          return
+        }
+
+        middlewaresConfig.routes = middlewaresConfig.routes?.map((route) => {
+          return {
+            ...route,
+            method: route.method ?? "USE",
+          }
+        })
+
+        const descriptor: GlobalMiddlewareDescriptor = {
+          config: middlewaresConfig,
+        }
+
+        this.validateMiddlewaresConfig(descriptor)
+
+        this.globalMiddlewaresDescriptor = descriptor
+      })
+    } catch (error) {
+      log({
+        activityId: this.activityId,
+        message: `Failed to load middleware configuration in ${absolutePath}. Skipping middleware configuration.`,
+      })
+
+      return
+    }
+  }
+
+  protected async createRoutesMap({
+    dirPath,
+    parentPath,
+  }: {
+    dirPath: string
+    parentPath?: string
+  }): Promise<void> {
+    await Promise.all(
+      await readdir(dirPath, { withFileTypes: true }).then((entries) => {
+        return entries
+          .filter((entry) => {
+            /**
+             * We exclude all files starting with an underscore
+             * as underscore is used to indicate that files are
+             * to be ignored.
+             */
+            if (entry.name.startsWith("_")) {
+              return false
+            }
+
+            const fullPath = join(dirPath, entry.name)
+
+            if (
+              this.excludes.length &&
+              this.excludes.some((exclude) => exclude.test(fullPath))
+            ) {
+              return false
+            }
+
+            // Get entry name without extension
+            const name = entry.name.replace(/\.[^/.]+$/, "")
+
+            if (entry.isFile() && name !== ROUTE_NAME) {
+              return false
+            }
+
+            return true
+          })
+          .map(async (entry) => {
+            const childPath = join(dirPath, entry.name)
+
+            if (entry.isDirectory()) {
+              return this.createRoutesMap({
+                dirPath: childPath,
+                parentPath: parentPath ?? dirPath,
+              })
+            }
+
+            return this.createRoutesDescriptor({
+              childPath,
+              parentPath,
+            })
+          })
+          .flat(Infinity)
+      })
+    )
+  }
+
+  protected async registerRoutes(): Promise<void> {
+    const prioritizedRoutes = prioritize([...this.routesMap.values()])
+
+    for (const descriptor of prioritizedRoutes) {
+      if (!descriptor.config?.routes?.length) {
+        continue
+      }
+
+      const routes = descriptor.config.routes! as RouteConfig[]
+
+      if (descriptor.config.shouldRequireAuth) {
```

Comment:
suggestion: base on the previous comment about auth, we could apply the customerAuthenticate as well except that this one is false by default. wdyt?